### PR TITLE
Refactor snapshot code

### PR DESCRIPTION
**Pull request by meghnasiingh**

### CODE REVIEW

**Review by sameshai — APPROVED** [review]

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: block/provider/create_volume.go [hunk 1]
```diff
@@ -79,6 +79,10 @@ func (vpcs *VPCSession) CreateVolume(volumeRequest provider.Volume) (volumeRespo
 
 	if err != nil {
 		vpcs.Logger.Debug("Failed to create volume from VPC provider", zap.Reflect("BackendError", err))
+		modelError, ok := err.(*models.Error)
+		if ok && len(modelError.Errors) > 0 && string(modelError.Errors[0].Code) == "snapshot_id_not_found" {
+			return nil, userError.GetUserError("StorageFindFailedWithSnapshotId", err)
```
Comment (arahamad) [inline block/provider/create_volume.go:84]: replace StorageFindFailedWithSnapshotId with SnapshotIDNotFound

File: block/provider/create_volume.go [hunk 2]
```diff
@@ -79,6 +79,10 @@ func (vpcs *VPCSession) CreateVolume(volumeRequest provider.Volume) (volumeRespo
 
 	if err != nil {
 		vpcs.Logger.Debug("Failed to create volume from VPC provider", zap.Reflect("BackendError", err))
+		modelError, ok := err.(*models.Error)
+		if ok && len(modelError.Errors) > 0 && string(modelError.Errors[0].Code) == "snapshot_id_not_found" {
```
Comment (arahamad) [inline block/provider/create_volume.go:83]: make a const for this hardcoded values

File: block/provider/delete_snapshot.go [hunk 3]
```diff
@@ -46,6 +46,10 @@ func (vpcs *VPCSession) DeleteSnapshot(snapshot *provider.Snapshot) error {
 	})
 
 	if err != nil {
+		modelError, ok := err.(*models.Error)
+		if ok && len(modelError.Errors) > 0 && string(modelError.Errors[0].Code) == "snapshots_not_found" {
```
Comment (arahamad) [inline block/provider/delete_snapshot.go:50]: lets create a const for all hardcoded values

File: block/provider/delete_snapshot.go [hunk 4]
```diff
@@ -46,6 +46,10 @@ func (vpcs *VPCSession) DeleteSnapshot(snapshot *provider.Snapshot) error {
...
+			return userError.GetUserError("StorageFindFailedWithSnapshotId", err)
```
Comment (arahamad) [inline block/provider/delete_snapshot.go:51]: replace StorageFindFailedWithSnapshotId with SnapshotIDNotFound

File: block/provider/create_volume.go (hunk 1 above)
Comment (meghnasiingh) [reply block/provider/create_volume.go]: Done

File: block/provider/create_volume.go (hunk 2 above)
Comment (meghnasiingh) [reply block/provider/create_volume.go]: Done

File: block/provider/delete_snapshot.go (hunk 3 above)
Comment (meghnasiingh) [reply block/provider/delete_snapshot.go]: Done

File: block/provider/delete_snapshot.go (hunk 4 above)
Comment (meghnasiingh) [reply block/provider/delete_snapshot.go]: Done

**Review by arahamad — APPROVED** [review]
lgtm

**Pull request merged**